daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/SeparatedSequenceUnparsers.scala

```diff
@@ -137,16 +142,21 @@ class OrderedSeparatedSequenceUnparser(
   }
 
   /**
-   * Unparses the separator only.
+   * Unparses just the separator, as well as any mandatory text alignment if it necessary
    *
    * Does not deals with infix boundary condition.
    */
   private def unparseJustSeparator(state: UState): Unit = {
+    if (sepMtaUnparserMaybe.isDefined) {
```

Comment:
This here is the crux of the change I think. You are laying down an alignment unparser for the separator, only if the optimizer has decided one could be needed, and it is laid down unconditionally.  Probably worth inserting comment to that affect here.